src/main/java/net/snowflake/client/jdbc/SnowflakePreparedStatementV1.java

```diff
@@ -436,7 +441,11 @@ public void setObject(int parameterIndex, Object x) throws SQLException {
   public boolean execute() throws SQLException {
     logger.debug("execute: {}", sql);
 
-    return executeInternal(sql, parameterBindings);
+    boolean success = executeInternal(sql, parameterBindings);
+    if (statementMetaData.isValidMetaData()) {
+      alreadyDescribed = true;
```

Comment:
Could you please test using multiple results and make sure alreadyDescribed is clear in between results.

Reply:
Sure! Added in assert statements in current tests for preparedStatement.isAlreadyDescribed() and added an additional test with multiple resultsets as well.